File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/AccountTypeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AccountTypeRow } from "./AccountTypeRow";

export interface AccountTypeColumns {
    AccountTypeId: Column<AccountTypeRow>;
    Description: Column<AccountTypeRow>;
}

export class AccountTypeColumns extends ColumnsBase<AccountTypeRow> {
    static readonly columnsKey = 'Sales.AccountType';
    static readonly Fields = fieldsProxy<AccountTypeColumns>();
}

File: 2026/Company/Company.Web/Modules/Sales/AccountType/AccountTypeColumns.cs
namespace Company.Sales.Columns;

[ColumnsScript("Sales.AccountType")]
[BasedOnRow(typeof(AccountTypeRow), CheckNames = true)]
public class AccountTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int AccountTypeId { get; set; }
    [EditLink]
    public string Description { get; set; }
}